File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
from collections import deque
def solution(arr):
    answer = []
    queue = deque()
    queue.append(arr[0])
    for i in range(1, len(arr)):
        temp = queue.pop()
        if temp == arr[i]:
            queue.append(temp)
        else:
            queue.append(temp)
            queue.append(arr[i])
    for i in queue:
        answer.append(i)
    return answer


print(solution([1,1,3,3,0,1,1]))

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        List<Integer> list = new ArrayList<>();
        Queue<Integer> queue = new LinkedList<>();
        
        for (int a: arr) {
            queue.add(a);
        }
        
        list.add(queue.peek());
        
        for (int num: queue) {
            if (list.get(list.size()-1) == num) {
                continue;
            }else{
                list.add(num);
            }
        }
        
        answer = new int[list.size()];
        
        for (int i=0; i<list.size(); i++) {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}